### PR TITLE
Added `resnet_jax.ipynb` for `resnet_torch.ipynb`

### PR DESCRIPTION
**Pull request by codeboy5**
## Description
Converted `resnet_torch.ipynb` notebook to `resnet_jax.ipynb`.

## Checklist:
 * [x]  Converted `torch` code to `jax`.
 * [x]  Verified that it runs on colab.
 
 ## Potential problems/Important remarks
 The best accuracy on the test set is same as the pytorch code. 

The notebook is available at https://colab.research.google.com/github/codeboy5/probml-notebooks/blob/add-resnet-jax/notebooks-d2l/resnet_jax.ipynb

### CODE REVIEW

**Comment by codeboy5** [comment]
Can you please review my code? @murphyk @mjsML @gerdm

**Comment by murphyk** [comment]
Looks great!
